src/System.Management.Automation/engine/runtime/Binding/Binders.cs

```diff
@@ -6769,6 +6793,45 @@ internal static Expression InvokeMethod(MethodBase mi, DynamicMetaObject target,
             List<Expression> initTemps = new List<Expression>();
             List<Expression> copyOutTemps = new List<Expression>();
 
+            ConstructorInfo constructorInfo = null;
+            MethodInfo methodInfo = mi as MethodInfo;
+            if (methodInfo != null)
```

Comment:
We could use C# 7.0 syntax.

Reply:
I tried the `is` syntax first and got a parsing error in the code below where `methodInfo` is used: `Use of unassigned local variable 'methodInfo'`. So I change it back to `as` syntax.